### DISCUSSION TITLE
--only-downloads doesn't download required msys for packages using vcpkg_fixup_pkgconfig

### DISCUSSION THREAD

**Discussion by trebconnell** [microsoft/vcpkg#22006] [Ideas]
**Describe the bug**
--only-downloads doesn't download msys for packages which use vcpkg_fixup_pkgconfig.  Then when I try to install with --no-downloads I get an error because msys is missing.

I tested with gtest and zstd, and they both reproduce this.

**Environment**
- OS: Windows
- Compiler: VS2022
package: gtest[core]:x86-windows -> 1.11.0#3
  vcpkg-tool version: 2021-12-09-724cea8b33cbf06c645f5095fa29773697da9761
  vcpkg-scripts version: 48bba4f9c 2021-12-10 (2 days ago)

**To Reproduce**
1. ./vcpkg install gtest --only-downloads
2. ./vcpkg install gtest --no-downloads
3. See error:
```
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:251 (message):
  Downloads are disabled, but
  'E:/devl/vcpkg/downloads/msys-mingw-w64-i686-pkg-config-0.29.2-2-any.pkg.tar.zst'
  does not exist.
Call Stack (most recent call first):
  scripts/cmake/vcpkg_acquire_msys.cmake:84 (vcpkg_download_distfile)
  scripts/cmake/vcpkg_acquire_msys.cmake:174 (z_vcpkg_acquire_msys_download_package)
  scripts/cmake/vcpkg_find_acquire_program.cmake:551 (vcpkg_acquire_msys)
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:198 (vcpkg_find_acquire_program)
  ports/gtest/portfile.cmake:49 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:142 (include)
```

**Expected behavior**
Error should not be encountered, and gtest should successfully install

**Failure logs** 
[msysdownloadfail.log](https://github.com/microsoft/vcpkg/files/7699989/msysdownloadfail.log)

**Comment by LilyWangLL** [comment]
Because vcpkg doesn't support use external msys package, so use `--no-downloads` will appear this error.

**Comment by dg0yt** [comment]
These downloads are needed *after* the build (for `vcpkg_fixup_pkgconfig`). But the current download-only implementation stops at the first actual build command. So these downloads are not reached.

**Comment by autoantwort** [comment]
I suggest using [asset caching](https://github.com/microsoft/vcpkg/blob/master/docs/users/assetcaching.md) to a local folder to reach your goal.

**Comment by fabienfl-orc** [comment]
Is there any way to avoid msys dependency or `vcpkg_fixup_pkgconfig` ?

**Reply by autoantwort** [reply]
No

**Reply by dg0yt** [reply]
IIUC you can avoid it by providing a pkg-config tool via environment variable `PKG_CONFIG`. A suitable implementation is `tools/pkgconf/pkgconf[.exe]` from port `pkgconf`.
